move/interchain_token_service/sources/types/token_manager_types.move

```diff
@@ -21,4 +26,10 @@ module interchain_token_service::token_manager_types {
     public fun mint_burn(): u8 {
         MINT_BURN
     }
+
+    public fun should_have_treasry_cap_for_link_token(token_manager_type: u8): bool {
```

Comment:
separate this into two checks, first convert raw u8/u256 to TokenManagerType for validation, and then check if that equals `TokenManagerType::mint_burn()` or native interchain token in the caller